prometheus_client/exposition.py

```diff
@@ -143,9 +143,16 @@ class ThreadingWSGIServer(ThreadingMixIn, WSGIServer):
     daemon_threads = True
 
 
+def _get_best_family(address, port):
+    infos = socket.getaddrinfo(address, port)
+    family, _, _, _, sockaddr = next(iter(infos))
+    return family, sockaddr[0]
+
+
 def start_wsgi_server(port, addr='', registry=REGISTRY):
     """Starts a WSGI server for prometheus metrics as a daemon thread."""
     app = make_wsgi_app(registry)
+    ThreadingWSGIServer.address_family, addr = _get_best_family(addr, port)
```

Comment:
Rather than update the address_family on the class, could we make a new class variable in this function and change it on that?

Reply:
I just did what http/server.py does as a good monkey but they certainly don't care about what kind of side effects they have for `python -m http` invocations.
I'm however not familiar with how this part of python works -- I just tried assigning the class to a new variable and that doesn't copy it e.g. this doesn't work:
```
      LocalServer = ThreadingWSGIServer
      LocalServer.address_family, addr = _get_best_family(addr, port)
      print(addr, LocalServer.address_family, ThreadingWSGIServer.address_family)
      httpd = make_server(addr, port, app, LocalServer, handler_class=_SilentHandler)
```
will print AF_INET6 twice or AF_INET twice depending on what was set.

stackoverflow has some horrible things with type() but I'm not sure this is worth the effort/risk unless you know how to do that reliably

Reply:
:(, I will try taking a look when I get a chance. My fear with this solution is if someone sets up multiple servers the address_family would be shared between them and cause issues just as you saw here.

Reply:
I've just checked though calling `start_http_server` with different address families in a row works fine, e.g.:
```
    start_http_server(8000, addr='::1')
    start_http_server(8001, addr='127.0.0.1')
```
will work and listen to both as expected

Since these are started off the main thread in most cases, we could just always restore the class address_family to AF_INET afer we're done creating the server -- would that be acceptable?
If subservers are created in another thread though this would be racy with whatever happens in other threads, but I don't think that's quite common.. Not much way of knowing though.

Reply:
Having a library that effectively mutates global state just does not feel right to me. Restoring the address_family might be ok, but as you point out the thread safety that worries me. Certainly it is an edge case, but would be hard to debug if something does go wrong. I will think about this some more/try some things when I have time.